adempiere/adempiere
# #1699 Add automatic reconciliation

Hi everybody, I am happy for share this new contribution for automation of bank statement.

Now is very easy make a bank statement from import:

Technical info: With it pull request is supported follow standard bank statement info:
 - [QIF](https://en.wikipedia.org/wiki/Quicken_Interchange_Format): Quicken Interchange Format
 - [MT940](https://en.wikipedia.org/wiki/MT940)
 - [OFX/OFC](https://en.wikipedia.org/wiki/Open_Financial_Exchange): Open Financial Exchange (older support)

Issues related:
 - fixes #1700
 - fixes #1701
 - fixes #1702
 - fixes #1699
 - fixes #1773
 - fixes #1706
 - fixes #1713
 - fixes #1807
 - fixes #1793

Forms Added:
 - Bank Statement Matcher

Process Added:
 - Create From Bank Statement for Bank Statement Matcher

Changes for Windows:
 - Add tab for relation of Bank Statement Matcher with Bank (Tab is added in Bank Definition)
 - Add Tab for sorting

Here a Step by Step for Functional Testing:

- Basic Setup for load and matching: 
  - Go to **System Rules** -> **Organization Rules** -> **Bank Statement Matcher**
![screenshot_20180816_001456](https://user-images.githubusercontent.com/2333092/44189871-e8e5d280-a0e9-11e8-802c-726cd4cee539.png)
  - Add new Bank Statement Matcher to it, here exist a example list:
![screenshot_20180816_001520](https://user-images.githubusercontent.com/2333092/44189867-e84d3c00-a0e9-11e8-8f2d-a63312241191.png)
  - Now, go to **System Rules** -> **Organization Rules** -> **Bank**
![screenshot_20180816_001538](https://user-images.githubusercontent.com/2333092/44189872-e8e5d280-a0e9-11e8-9f01-a8211668e77b.png)
  - Go to **Statement Matcher** tab and add custom matcher that apply for your bank:
![screenshot_20180816_001606](https://user-images.githubusercontent.com/2333092/44189868-e84d3c00-a0e9-11e8-918d-a25ab97f94ff.png)
  - Go to **Sequence** tab and order priority for matching:
![screenshot_20180816_001628](https://user-images.githubusercontent.com/2333092/44189869-e8e5d280-a0e9-11e8-9606-4cb73de6f215.png)
  - Now, you can go to **Statement Loader** tab and define custom loader for bank statement
![screenshot_20180816_001756](https://user-images.githubusercontent.com/2333092/44189876-e97e6900-a0e9-11e8-810b-3f3a2e2d9a79.png)
- Load **Bank Statement**:
  - Go to **System Admin** -> **Data** -> **Data Import** -> **Load Bank Statement**
![screenshot_20180816_001705](https://user-images.githubusercontent.com/2333092/44189873-e8e5d280-a0e9-11e8-9b42-3f1dcf93dbe2.png)
  - Select **Bank Statement Loader** and and **file to load** parameters and run
![screenshot_20180816_001925](https://user-images.githubusercontent.com/2333092/44189870-e8e5d280-a0e9-11e8-8e4e-8a225fcb2fe2.png)
- Make matching:
  - Go to **Open Items** -> **Bank Statement**
![screenshot_20180816_001949](https://user-images.githubusercontent.com/2333092/44189874-e97e6900-a0e9-11e8-986c-6127b5419395.png)
  - Into **Bank Statement** window go to **Process Run** and select  **Bank Statement Match** option
![screenshot_20180816_002003](https://user-images.githubusercontent.com/2333092/44189877-e97e6900-a0e9-11e8-877b-442df79225b5.png)
  - When form is opened, default value is setted for bank account (Now you can press **Refresh** button!!!):
![screenshot_20180816_002105](https://user-images.githubusercontent.com/2333092/44189878-ea16ff80-a0e9-11e8-87a7-b3a7498f9823.png)
  - When it is loaded you can see all data:
    - Left: **Imported Payments**
    - Right: **System Payments**
    - Below: **Matched Payments**
  - Now you can press **Simulate Match** button
![screenshot_20180815_213718](https://user-images.githubusercontent.com/2333092/44189875-e97e6900-a0e9-11e8-9cd0-f7dcd568a2ef.png)
  - All matched payment are showed (Now press ok for add to Bank Statement) See result:
![screenshot_20180816_002635](https://user-images.githubusercontent.com/2333092/44189880-eaaf9600-a0e9-11e8-9dc5-00529ada9093.png)

Note: When payments are add to **Bank Statement** they are imported automatically.

Best regard
